datumaro/components/errors.py

```diff
@@ -223,14 +225,35 @@ def _my__init__(self, msg=None, *, sources=None):
 
 @attrs
 class MismatchingImageInfoError(DatasetMergeError):
-    item_id = attrib()
-    a = attrib()
-    b = attrib()
+    item_id: Optional[Tuple[str, str]] = attrib()
+    a: int = attrib()
+    b: int = attrib()
 
     def __str__(self):
         return "Item %s: mismatching image size info: %s vs %s" % \
             (self.item_id, self.a, self.b)
 
+@attrs
+class MismatchingImagePathError(DatasetMergeError):
+    item_id: Optional[Tuple[str, str]] = attrib()
+    a: str = attrib()
+    b: str = attrib()
+
+    def __str__(self):
+        return "Item %s: mismatching image path info: %s vs %s" % \
+            (self.item_id, self.a, self.b)
+
+@attrs
+class MismatchingAttributesError(DatasetMergeError):
+    item_id: Optional[Tuple[str, str]] = attrib()
+    key: str = attrib()
+    a: Any = attrib()
+    b: Any = attrib()
+
+    def __str__(self):
+        return "Item %s: mismatching image attribute %s: %s vs %s" % \
+            (self.item_id or '', self.key, self.a, self.b)
```

Comment:
Maybe, but I don't want to be very specific at this point about exceptions. Probably, it can be designed more thoroughly in a separate PR. For this PR I'd prefer to make the `ExactMerge` methods private to avoid `item_id=None`.

Reply:
Works for me.